### PR TITLE
Replace wx/xml with pugixml in rXML

### PR DESCRIPTION
**Pull request by mpm11011**
`git reset --hard upstream/master` automatically closed the previous PR. At least now its clean.

### CODE REVIEW

**Comment by danilaml** [comment]
Looks good now. Btw is there some easy way to check if it's working correctly?

**Comment by mpm11011** [comment]
By working correctly I assume you mean the XML change. In that respect,
since it  is only used by trophy files, I think you have to receive a
trophy, access a trophy or trophy list, etc.

If by working working correctly you mean does it compile, only time and the
build bots will tell.

**Comment by danilaml** [comment]
@mpm11011 yeah, trophys weren't really working before and I'm not sure that xml was the problem. I wonder if it fixes anything.

**Comment by mpm11011** [comment]
The change was to remove wx dependencies in emucore. The Utilities folder
is included. In reality, it probably won't change anything other than our
reliance on the wx library.

**Comment by mpm11011** [comment]
@vlj You asked me if I was ready to merge, and now it mostly definitely is.

File: Utilities/rXml.cpp [hunk 1]
```diff
@@ -1,117 +1,120 @@
...
 std::string rXmlNode::GetName()
 {
-	return fmt::ToUTF8(reinterpret_cast<wxXmlNode*>(handle)->GetName());
+	return handle->name();
 }
 
 std::string rXmlNode::GetAttribute(const std::string &name)
 {
-	return fmt::ToUTF8(reinterpret_cast<wxXmlNode*>(handle)->GetAttribute(fmt::FromUTF8(name)));
+	auto pred = [name](pugi::xml_attribute attr) { return (name == attr.name()); };
```
Comment (danilaml) [inline Utilities/rXml.cpp]: `name` could probably be captured by reference here

Comment (mpm11011) [inline Utilities/rXml.cpp]: `[&name]` like that, correct?

Comment (danilaml) [inline Utilities/rXml.cpp]: yes.

**Comment by danilaml** [comment]
Needs a rebase before merge.

**Comment by mpm11011** [comment]
Done and done.

**Comment by danilaml** [comment]
@mpm11011 It'd better if you left `Pugixml to replace wx/xml in rXML` as a name of the commit (you can see how it's easily done in git [here](https://help.github.com/articles/changing-a-commit-message/)).

**Comment by mpm11011** [comment]
Whoops. I thought the merge would take the title of the PR as its message. Fixed.

**Comment by mpm11011** [comment]
Yes, #1665 will fix this. My apologies.

**Pull request merged**